3-projects/cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,267 @@
+const express = require("express");
+const app = express();
+app.use(express.json());
+
+const { Pool } = require("pg");
+
+const pool = new Pool({
+  user: "asif",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "786god",
+  port: 5432,
+});
+
+app.get("/customers", function (req, res) {
+  pool
+    .query("SELECT * FROM customers")
+    .then((result) => {
+      if (result) {
+        res.status(200).json(result.rows);
+      } else res.status(400).send("Couldn't retrieve");
+    })
+    .catch((e) => console.error(e));
+});
+app.get("/customers/:customerId/orders", function (req, res) {
+  const customerId = req.params.customerId;
+
+  const query =
+    "SELECT o.order_reference, o.order_date, p.product_name, pa.unit_price, s.supplier_name, oi.quantity  FROM orders o INNER JOIN order_items oi ON o.id = oi.order_id INNER JOIN products p ON oi.product_id = p.id INNER JOIN product_availability pa ON oi.product_id = pa.prod_id INNER JOIN suppliers s ON oi.supplier_id = s.id WHERE customer_id = $1";
+
```

Comment:
You need to join order items with product availability on both the product id and the supplier id.